File: NoDataView/ViewController.m
//
//  ViewController.m
//  NoDataView
//
//  Created by LvJianfeng on 16/4/18.
//  Copyright © 2016年 LvJianfeng. All rights reserved.
//

#import "ViewController.h"
#import "LLNoDataView.h"

@interface ViewController () <LLNoDataViewTouchDelegate>
@property (strong, nonatomic) NSArray *data;
@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view, typically from a nib.
    //没有数据
    LLNoDataView *dataView = [[LLNoDataView alloc] initNoDataWithFrame:self.tableView.bounds description:@"没有数据" canTouch:YES];
    dataView.delegate = self;
    self.tableView.tableHeaderView = dataView;
    
    //实例一次，再次修改提示文本信息
    dataView.tipLabel.text = @"没有搜索到\"LLNoDataView\"的数据";
    
//    //没有网络
//    LLNoDataView *dataView = [[LLNoDataView alloc] initNoInternetWithFrame:self.tableView.bounds description:@"网络连接失败" canTouch:YES];
//    dataView.delegate = self;
//    self.tableView.tableHeaderView = dataView;
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView{return 1;}

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section{return self.data.count;}

- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath{
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"TestCell"];
    if (!cell) {
        cell = [[UITableViewCell alloc] initWithStyle:UITableViewCellStyleDefault reuseIdentifier:@"TestCell"];
    }
    cell.textLabel.text = [NSString stringWithFormat:@"第%ld行",indexPath.row];
    return cell;
}

- (NSArray *)data{
    if (!_data) {
        _data = [[NSArray alloc] init];//WithObjects:@"",@"", nil
    }
    return _data;
}

- (void)didTouchLLNoDataView{
    self.data = @[@"",@"",@"",@"",@"",@"",@"",@""];
    [self.tableView reloadData];
    self.tableView.tableHeaderView = nil;
}
@end

File: Swift版本/LLNoDataViewSwift/LLNoDataViewSwift/ViewController.swift
//
//  ViewController.swift
//  LLNoDataViewSwift
//
//  Created by LvJianfeng on 2016/12/20.
//  Copyright © 2016年 LvJianfeng. All rights reserved.
//
//  https://github.com/LvJianfeng/LLNoDataView

import UIKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // 使用更简单了
        
        // Demo 1
        let demo1 = LLNoDataView.ll_NoData(frame: self.view.bounds)
        
        // Demo 2
        let demo2 = LLNoDataView.ll_NoInternet(frame: self.view.bounds)
        
        // Demo 3
        let demo3 = LLNoDataView.ll_NoInternet(frame: self.view.bounds, reloadHandler: {
            print("刷新")
        })
        self.view.addSubview(demo3)
        
        // Demo 4
//        let demo4 = LLNoDataView.ll_NoInternet(frame: CGRect.init(x: 0, y: 0, width: 375, height: 320), backgroundColor: UIColor.groupTableViewBackground, reloadHandler: {
//        
//        })
//        self?.tableView.tableFooterView = LLNoDataView.ll_NoData(frame: CGRect.init(x: 0, y: 0, w: s_SCREEN_WIDTH, h: s_SCREEN_HEIGHT-64-45))
        
//        let demo5 = LLNoDataView.ll_NoInternet(frame: CGRect.init(x: 0, y: 0, width: 375, height: 320), backgroundColor: UIColor.groupTableViewBackground, reloadHandler: {
//
//        })
//        self?.tableView.tableHeaderView = LLNoDataView.ll_NoData(frame: CGRect.init(x: 0, y: 0, w: s_SCREEN_WIDTH, h: s_SCREEN_HEIGHT-64-45))
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

